microsoft/testsuites/core/infiniand.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from assertpy import assert_that
+
+from lisa import Logger, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import Ubuntu
+from lisa.util import SkippedException
+
+
+@TestSuiteMetadata(
+    area="core",
+    category="functional",
+    description="""
+    Tests the functionality of infiniband.
+    """,
+)
+class Infiniband(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Determine whether the VM is IB over ND, IB over SR-IOV, or non-HPC
+        2. Ensure waagent is configures with OS.EnableRDMA=y
+        3. Check that appropriate drivers are present
+        """,
+        priority=2,
+    )
+    def verify_hpc_vm(self, log: Logger, node: Node) -> None:
+
+        # Ubuntu is used for HPC SKU but manual configuration required.
+        # It's exceptional.
+        if isinstance(node.os, Ubuntu):
+            raise SkippedException(
+                f"{node.os.name} not supported. "
+                "Only HPC machines are supported by this test."
+            )
+
+        if node.execute("test -d /sys/class/infiniband", shell=True).exit_code != 0:
```

Comment:
this one is decided by VM size, not image type

Reply:
``` Bash
# Ubuntu is used for HPC SKU but manual configuration required. It's exceptional.
function is_hpc_vm() {
	GetDistro
	if [ -d /sys/class/infiniband/ ] && [[ $DISTRO != "ubuntu"* ]]; then
```
This is the logic from v2 that I used. The test is skipped for all Ubuntu distros and the existence of the infiniband directory is used to determine whether the machine is HPC. If you have a better way of determining HPC SKU, please let me know.